### ISSUE TITLE
Support mysql 8.0

### ISSUE THREAD

**Issue by bes-internal** [major/MySQLTuner-perl#485]
1.
```
-------- Security Recommendations ------------------------------------------------------------------
[OK] There are no anonymous accounts for any database users
[!!] failed to execute: SELECT CONCAT(user, '@', host) FROM mysql.user WHERE (password = '' OR password IS NULL) AND plugin NOT IN ('unix_socket', 'win_socket')
[!!] FAIL Execute SQL / return code: 256
[OK] All database users have passwords assigned
[!!] failed to execute: SELECT CONCAT(user, '@', host) FROM mysql.user WHERE CAST(password as Binary) = PASSWORD(user) OR CAST(password as Binary) = PASSWORD(UPPER(user)) OR CAST(password as Binary) = PASSWORD(CONCAT(UPPER(LEFT(User, 1)), SUBSTRING(User, 2, LENGTH(User))))
[!!] FAIL Execute SQL / return code: 256
[...]
```
password --> authentication_string

2.
```
Variables to adjust:
    query_cache_type (=0)
```
query_cache_* removed from mysql 8.0

**Comment by jmrenouard** [comment]
Hi @bes-internal 

Two issues here:
1° remove query_cache for MySQL 8.x
2° handle credentials checks in a a rigth way for MySQL 8.x

**Comment by grooverdan** [comment]
query cache resolved in b8ef183b, so just credential checking around the default caching_sha2_password plugin needed.
